File: Marcianitos/src/deepspace/Loot.java
package deepspace;

/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */

/**
 *
 * @author mario
 */
class Loot {
    
    private int nSupplies;
    private int nWeapons;
    private int nShields;
    private int nHangars;
    private int nMedals;
    
    Loot(int supplies, int weapons, int shields, int hangars, int medals){
        
        nSupplies=supplies;
        nWeapons=weapons;
        nShields=shields;
        nHangars= hangars;
        nMedals=medals;       
       
    }
    
    LootToUI getUIVersion() {
        return new LootToUI(this);
    }
    
    public int getNSupplies(){
        
        return nSupplies;
    }
    
    public int getNWeapons(){
        
        return nWeapons;
    }
    
    public int getNShields(){
        
        return nShields;
    }
    
    public int getNHangars(){
        
        return nHangars;
    }
     
    public int getNMedals(){
        
        return nMedals;
    }
    
    public String toString(){
        
        String salida;
        
        salida = "Supplies=" + nSupplies + ", Weapons=" + nWeapons + ", Shields=" + nShields + ", Hangars=" + nHangars;

        return salida;
    }
    
}

File: RUBY/src/Loot.rb
#encoding: UTF-8
require_relative 'LootToUI'

module Deepspace
	class Loot	#terminada
		def initialize (supplies,weapons,shields,hangars,medals,ef,city)  # son todos int menos ef y city q son bool
			@nSupplies = supplies
			@nWeapons = weapons
			@nShields = shields
			@nHangars = hangars
			@nMedals = medals
			
			@efficient = ef
			@spaceCity = city
		end

		attr_reader :nSupplies, :nWeapons, :nShields, :nHangars, :nMedals, :efficient, :spaceCity
		
		def getUIversion()
			return LootToUI.new(self)
		end
		
		def to_s
			return "Supplies: #{@nSupplies}, Weapons: #{@nWeapons}, Shields: #{@nShields}, Hangars: #{@nHangars}, Medals: #{@nMedals}, getEfficient: #{@getEfficient}, spaceCity: #{@spaceCity}"
		end
	end

end
